### ISSUE TITLE
Quay and Nooba, .spec.ObjectBucketName: field not declared in schema

### ISSUE THREAD

**Issue by a1ex-var1amov** [quay/quay-operator#465]
Hello there,

I'm facing the following issue trying to deploy fully managed QuayRegistry using OCS/ODF based Nooba:

```
2021-05-25T12:15:20.926Z        INFO    controllers.QuayRegistry        finished creating/updating object       {"quayregistry": "openshift-operators", "Name": "skynet-quay-app", "GroupVersionKind": "autoscaling/v2beta2, Kind=HorizontalPodAutoscaler"}
2021-05-25T12:15:20.926Z        INFO    controllers.QuayRegistry        creating/updating object        {"quayregistry": "openshift-operators", "Name": "skynet-quay-datastore", "GroupVersionKind": "objectbucket.io/v1alpha1, Kind=ObjectBucketClaim"}
2021-05-25T12:15:20.936Z        ERROR   controllers.QuayRegistry        failed to create/update object  {"quayregistry": "openshift-operators", "Name": "skynet-quay-datastore", "GroupVersionKind": "objectbucket.io/v1alpha1, Kind=ObjectBucketClaim", "error": "failed to create typed patch object: .spec.ObjectBucketName: field not declared in schema"}
github.com/go-logr/zapr.(*zapLogger).Error
        /workspace/vendor/github.com/go-logr/zapr/zapr.go:128
github.com/quay/quay-operator/controllers/quay.(*QuayRegistryReconciler).createOrUpdateObject
        /workspace/controllers/quay/quayregistry_controller.go:524
github.com/quay/quay-operator/controllers/quay.(*QuayRegistryReconciler).Reconcile
        /workspace/controllers/quay/quayregistry_controller.go:290
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).reconcileHandler
        /workspace/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:245
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem
        /workspace/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:221
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).worker
        /workspace/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:200
k8s.io/apimachinery/pkg/util/wait.JitterUntil.func1
        /workspace/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:152
k8s.io/apimachinery/pkg/util/wait.JitterUntil
        /workspace/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:153
k8s.io/apimachinery/pkg/util/wait.Until
        /workspace/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:88
2021-05-25T12:15:21.946Z        DEBUG   controller-runtime.controller   Successfully Reconciled {"controller": "quayregistry", "request": "openshift-operators/skynet"}

2021-05-25T12:15:21.946Z        DEBUG   controller-runtime.manager.events       Warning {"object": {"kind":"QuayRegistry","namespace":"openshift-operators","name":"skynet","uid":"fd330e28-d29b-4fbf-b83d-65cd171a98f8","apiVersion":"quay.redhat.com/v1","resourceVersion":"1140551"}, "reason": "ComponentCreationFailed", "message": "all Kubernetes objects not created/updated successfully: failed to create typed patch object: .spec.ObjectBucketName: field not declared in schema"}
```

The same for any other QuayRegistry configurations. 
Am I missing some prerequisites?

UPD: 
- OCP 4.7.11
- OCS 4.7.0
- RH Quay 3.5.1

Thank you!

**Comment by marouim** [comment]
Same issue here.. ODF 4.7 and Quay 3.5 .. Same issue with 3.4 operator.

**Comment by dmesser** [comment]
Hi, yes, something seems wrong at the API level, the schema of the ObjectBucketClaim API changed slightly but only superficially. A workaround exists by setting `objectstorage` component to `managed: false`, manually applying the [manifest](https://github.com/quay/quay-operator/blob/master/kustomize/components/objectstorage/quay-datastore.objectbucketclaim.yaml) and taking the credentials from the status section of the claim and entering in the Quay config editor.

**Comment by a1ex-var1amov** [comment]
@dmesser Hi Daniel, thank you! Are there any specific reasons to use the manifest you mentioned? Can I go with some random ObjectBucketClaim?

**Comment by PiotrKlimczak** [comment]
Same issue

**Comment by dmesser** [comment]
@a1ex-var1amov no this was just the closest example.

**Comment by kcalliga** [comment]
Is there any update to this? Any related Bugzilla?

**Comment by dmesser** [comment]
@kcalliga The issue has been addressed here - https://github.com/quay/quay-operator/pull/466

We track the releases in Jira and this one would be a fix in 3.5.3 and upcoming 3.6 as per https://issues.redhat.com/browse/PROJQUAY-2051

**Comment by flavianmissi** [comment]
As mentioned by @dmesser, this has been fixed in 3.5.3 which is now generally available (see https://issues.redhat.com/projects/PROJQUAY/versions/12358488). 
I'm closing this now - do reopen if needed.
